eclipseMavenCentral does not convert "org.eclipse.jdt.core.compiler.batch" correctly

Probably due to legacy reasons it was decided to publish bundle `org.eclipse.jdt.core.compiler.batch` as:

```xml
<dependency>
    <groupId>org.eclipse.jdt</groupId>
    <artifactId>ecj</artifactId>
    <version>${jdtVersion}</version>
</dependency>
```

---
This special mapping is defined in the `eclipse.platform.releng` project in the CBI aggregator configuration file:

```xml
<mavenMappings namePattern="(org\.eclipse\.jdt)\.core\.compiler\.batch" groupId="$1" artifactId="ecj"/>
```
Source: https://git.eclipse.org/c/platform/eclipse.platform.releng.git/tree/publish-to-maven-central/SDK4Mvn.aggr#n36

---

Currently `eclipseMavenCentral` is now aware of this special mapping, it is failing with:

```
> Task :my_project:compileJava FAILED

FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':my_project:compileJava'.
> Could not resolve all files for configuration ':my_project:compileClasspath'.
   > Could not find org.eclipse.jdt:org.eclipse.jdt.core.compiler.batch:3.20.0.
     Searched in the following locations:
       - https://repo.maven.apache.org/maven2/org/eclipse/jdt/org.eclipse.jdt.core.compiler.batch/3.20.0/org.eclipse.jdt.core.compiler.batch-3.20.0.pom
     Required by:
         project :my_project
```

--- a/CHANGES.md
+++ b/CHANGES.md
@@ -2,6 +2,9 @@
 
 ## [Unreleased]
 
+### Fixed
+- Fix ecj mapping in eclipseMavenCentral ([#158](https://github.com/diffplug/goomph/pull/158))
+
 ## [3.32.1] - 2021-09-09
 
 ### Fixed

--- a/src/main/java/com/diffplug/gradle/eclipse/MavenCentralMapping.java
+++ b/src/main/java/com/diffplug/gradle/eclipse/MavenCentralMapping.java
@@ -51,7 +51,9 @@ public class MavenCentralMapping {
 
 	/** Returns the MavenCentral groupId:artifactId appropriate for the given bundleId. */
 	public static String groupIdArtifactId(String bundleId) {
-		if (bundleId.startsWith(JDT)) {
+		if ("org.eclipse.jdt.core.compiler.batch".equals(bundleId)) {
+			return JDT + ":ecj";
+		} else if (bundleId.startsWith(JDT)) {
 			return JDT + ":" + bundleId;
 		} else if (bundleId.startsWith(PDE)) {
 			return PDE + ":" + bundleId;

